src/packagedcode/citation.py

```diff
@@ -0,0 +1,85 @@
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# ScanCode is a trademark of nexB Inc.
+# SPDX-License-Identifier: Apache-2.0
+# See http://www.apache.org/licenses/LICENSE-2.0 for the license text.
+# See https://github.com/nexB/scancode-toolkit for support or download.
+# See https://aboutcode.org for more information about nexB OSS projects.
+#
+
+import io
+import saneyaml
+from packagedcode import models
+
+"""
+Handle CITATION.cff files, see https://citation-file-format.github.io/
+https://github.com/citation-file-format/citation-file-format/blob/main/schema-guide.md
+
+See https://github.com/citation-file-format/citation-file-format/tree/main/examples/1.2.0/pass for examples.
+"""
+
+class CitationHandler(models.DatafileHandler):
+    datasource_id = 'citation_cff'
+    default_package_type = 'citation'
+    path_patterns = ('*/CITATION.cff',)

```

Comment:
Based on https://github.com/search?q=path%3A*.cff&type=Code&ref=advsearch&l=&l=&p=1  these are often also in lowercase and others, and as we do a case-sensitive matching [here](https://github.com/nexB/scancode-toolkit/blob/develop/src/packagedcode/models.py#L974 ) (should we? :thinking:) we would fail on lowercase and others, should specify more cases here.